client/components/charts/charts.directives.js

```diff
@@ -1216,9 +1216,11 @@ app.directive('heartbeatChart', ['$timeout', 'Report', '$routeParams', 'COLOURS'
     // ClientDetails.client.ap_mac = undefined;
 
     controller.$scope.$on('resizeClientChart', function (evt, type){
-      if (a) {
-        drawChart();
-      }
+      drawChart();
+    });
+
+    controller.$scope.$on('loadClientChart', function  (){
```

Comment:
At least on beta, I see the same mechanism used for at least 6 other charts. And I believe it is necessary to act on loadClientChart to be able to reload the data when the user changes the time period.
But if you think there's a better way to implement this behavior, please provide some details so I can make the appropriate changes.